sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/tests/TimeSeriesInsightsQuerySeriesTests.cs

```diff
@@ -0,0 +1,217 @@
+            TimeSeriesId tsiId = await GetUniqueTimeSeriesInstanceIdAsync(tsiClient, modelSettings.TimeSeriesIdProperties.Count)
+                .ConfigureAwait(false);
+
+            try
+            {
+                // Send some events to the IoT hub
+                await QueryTestsHelper.SendEventsToHubAsync(
+                        deviceClient,
+                        tsiId,
+                        modelSettings.TimeSeriesIdProperties.ToArray(),
+                        10)
+                    .ConfigureAwait(false);
+
+                // Act
+
+                // Query for temperature events with two calculateions. First with the temperature value as is, and the second
+                // with the temperature value multiplied by 2.
+                DateTimeOffset now = Recording.UtcNow;
+                DateTimeOffset endTime = now.AddMinutes(10);
+                DateTimeOffset startTime = now.AddMinutes(-10);
+
+                var temperatureNumericVariable = new NumericVariable(
+                    new TimeSeriesExpression($"$event.{Temperature}"),
+                    new TimeSeriesExpression("avg($value)"));
+                var temperatureNumericVariableTimesTwo = new NumericVariable(
+                    new TimeSeriesExpression($"$event.{Temperature} * 2"),
+                    new TimeSeriesExpression("avg($value)"));
+                var temperatureTimesTwoVariableName = $"{Temperature}TimesTwo";
+                var inlineVariables = new Dictionary<string, TimeSeriesVariable>
+                {
+                    [Temperature] = temperatureNumericVariable,
+                    [temperatureTimesTwoVariableName] = temperatureNumericVariableTimesTwo,
+                };
+
+                var querySeriesRequestOptions = new QuerySeriesRequestOptions
+                {
+                    InlineVariables = inlineVariables,
+                };
+
+                // This retry logic was added as the TSI instance are not immediately available after creation
+                await TestRetryHelper.RetryAsync<AsyncPageable<QueryResultPage>>(async () =>
+                {
+                    AsyncPageable<QueryResultPage> querySeriesEventsPages = tsiClient.QuerySeriesAsync(
+                        tsiId,
+                        startTime,
+                        endTime,
+                        querySeriesRequestOptions);
+
+                    await foreach (QueryResultPage seriesEventsPage in querySeriesEventsPages)
+                    {
+                        seriesEventsPage.Timestamps.Should().HaveCount(10);
+                        seriesEventsPage.Timestamps.Should().OnlyContain(timeStamp => timeStamp >= startTime)
+                        .And
+                         .OnlyContain(timeStamp => timeStamp <= endTime);
+                        seriesEventsPage.Properties.Count.Should().Be(3); // EventCount, Temperature and TemperatureTimesTwo
```

Comment:
eventcount is a default one right? Always present kind of thing?

Reply:
Yup. I believe that is the case when no properties are projected.